js/data/bucket/fill_bucket.js

```diff
@@ -55,71 +54,51 @@ class FillBucket extends Bucket {
         return fillInterfaces;
     }
 
-    addVertex(vertexArray, x, y) {
-        return vertexArray.emplaceBack(x, y);
-    }
-
     addFeature(feature) {
-        const lines = loadGeometry(feature);
-        const polygons = convertCoords(classifyRings(lines, EARCUT_MAX_RINGS));
-
-        this.factor = Math.pow(2, 13);
-
         const startGroup = this.prepareArrayGroup('fill', 0);
         const startIndex = startGroup.layoutVertexArray.length;
 
-        for (let i = 0; i < polygons.length; i++) {
-            this.addPolygon(polygons[i]);
-        }
+        for (const polygon of classifyRings(loadGeometry(feature), EARCUT_MAX_RINGS)) {
```

Comment:
I think this reduces clarity a bit — I'd prefer to have the part after `of` as a constant in similar places.